### PR TITLE
go-sqlsmith: support hints in select/update/delete statement

### PR DESCRIPTION
**Pull request by Illyrix**
### What problem does this PR solve? 
sqlsmith now supports hints
### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by mahjonp** [comment]
@Illyrix could you give me an example generated by SQLSmith with SQL hint supported?

**Comment by you06** [comment]
You need to run `make tidy`.

File: pkg/go-sqlsmith/builtin/hint.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	h.HintName = model.NewCIStr(hintKeyword.name)
+
+	if hintKeyword.maxArg == 0 {
+		return
+	}
+
+	if hintKeyword.maxArg == 1 {
+		switch hintKeyword.name {
+		case "use_toja", "enable_plan_cache", "use_cascades":
+			h.HintData = util.Rd(2)%2 == 1
```
Comment (you06) [inline pkg/go-sqlsmith/builtin/hint.go]: You may add a function named `RdBool` in `util/rand.go`.

File: pkg/go-sqlsmith/builtin/hint.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		case "memory_quota":
+			h.HintData = int64(util.Rd(40960000) + 10240000)
```
Comment (you06) [inline pkg/go-sqlsmith/builtin/hint.go]: Use `util.RdRange` instead, why do you choose this range?

File: pkg/go-sqlsmith/builtin/hint.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		case "max_execution_time":
+			h.HintData = uint64(util.Rd(1000) + 500)
```
Comment (you06) [inline pkg/go-sqlsmith/builtin/hint.go]: Use `util.RdRange` instead.

File: pkg/go-sqlsmith/builtin/hint.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	// not released?
+	{"query_type", 1, 1, false, false, false},
+
+	{"inl_hash_join", 1, -1, false, false, false},
+	{"inl_merge_join", 1, -1, false, false, false},
+}
+
+func GenerateHintExpr(table *types.Table) (h *ast.TableOptimizerHint) {
+	enabledKeywords := make([]*hintClass, len(hintKeywords))
+	copy(enabledKeywords, hintKeywords)
```
Comment (you06) [inline pkg/go-sqlsmith/builtin/hint.go]: Why not just use `enabledKeywords := hintKeywords`?

File: pkg/go-sqlsmith/stateflow/walker.go [hunk 5]
```diff
@@ -212,6 +221,22 @@ func (s *StateFlow) walkExprNode(node ast.ExprNode, table *types.Table, column *
...
+	for i := 0; i < hintLength; i++ {
+		if h := builtin.GenerateHintExpr(table); h != nil {
+			// remove duplicated hints
+			if _, ok := hintNames[h.HintName.String()]; !ok {
+				hList = append(hList, h)
+				hintNames[h.HintName.String()] = true
+			}
+		}
+	}
+	// TODO: remove conflict hints
```
Comment (you06) [inline pkg/go-sqlsmith/stateflow/walker.go]: What will be if there are conflict hints?

Comment (Illyrix) [reply pkg/go-sqlsmith/stateflow/walker.go]: onw of two confict hints may be ignored...

**Comment by mahjonp** [comment]
PTAL @you06

File: pkg/test-infra/fixture/fixture.go [hunk 6]
```diff
@@ -181,11 +182,12 @@ func init() {
 	flag.StringVar(&Context.ABTestConfig.TiDBConfigFile, "abtest.tidb-config", "", "tidb config file for cluster B")
 	flag.StringVar(&Context.ABTestConfig.TiKVConfigFile, "abtest.tikv-config", "", "tikv config file for cluster B")
 	flag.StringVar(&Context.ABTestConfig.PDConfigFile, "abtest.pd-config", "", "pd config file for cluster B")
-	flag.StringVar(&Context.ABTestConfig.ClusterBVersion, "abtest.image-version", "", "specify version for cluster B")
+	flag.StringVar(&Context.ABTestConfig.ClusterBVersion, "abtest.image-version", "latest", "specify version for cluster B")
 	flag.StringVar(&Context.ABTestConfig.LogPath, "abtest.log", "", "log path for abtest, default to stdout")
 	flag.IntVar(&Context.ABTestConfig.Concurrency, "abtest.concurrency", 3, "test concurrency, parallel session number")
 	flag.BoolVar(&Context.ABTestConfig.GeneralLog, "abtest.general-log", false, "enable general log in TiDB")
 
+	flag.BoolVar(&Context.EnableHint, "enable-hint", true, "enable to generate sql hint")
```
Comment (mahjonp) [inline pkg/test-infra/fixture/fixture.go]: use false as default value maybe better, because 3.1 and 3.0 branch don't support SQL HINT now.

**Pull request merged**